pkg/daemon/daemon.go

```diff
@@ -247,26 +252,27 @@ func (d *daemon) AddPeriodicUpdate() {
 			networks := podNetworksMap[pod.UID]
 			netAnnotations, err := json.Marshal(networks)
 			if err != nil {
+				failedPods = append(failedPods, pod)
 				glog.Warningf("AddPeriodicUpdate(): failed to dump networks %+v of pod into json with error: %v",
 					networks, err)
 				continue
 			}
 			pod.Annotations[v1.NetworkAttachmentAnnot] = string(netAnnotations)
 			if err := d.kubeClient.SetAnnotationsOnPod(pod, pod.Annotations); err != nil {
 				if !strings.Contains(strings.ToLower(err.Error()), "not found") {
+					failedPods = append(failedPods, pod)
```

Comment:
if it not found it mean that it was deleted, no need to add it to the failedPods

Reply:
I agree, the case here if it failed and it is not deleted

Reply:
ha ok I missed the not